docs/reference/text-structure/apis/find-field-structure.asciidoc

```diff
@@ -0,0 +1,309 @@
+However, you can optionally override some of the decisions about the text
+structure by specifying one or more query parameters.
+
+Details of the output can be seen in the <<find-field-structure-examples,examples>>.
+
+If the structure finder produces unexpected results for some text,
+specify the `explain` query parameter. It causes an `explanation` to appear in
+the response, which should help in determining why the returned structure was
+chosen.
+
+[discrete]
+[[find-field-structure-query-parms]]
+== {api-query-parms-title}
+
+`index`::
+(Required, string) The name of the index containing the field
+
+`field`::
+(Required, string) The name of the field that's analyzed.
+
+include::{es-repo-dir}/text-structure/apis/find-structure-shared.asciidoc[tag=param-column-names]
+include::{es-repo-dir}/text-structure/apis/find-structure-shared.asciidoc[tag=param-delimiter]
+include::{es-repo-dir}/text-structure/apis/find-structure-shared.asciidoc[tag=param-explain]
+include::{es-repo-dir}/text-structure/apis/find-structure-shared.asciidoc[tag=param-format]
+include::{es-repo-dir}/text-structure/apis/find-structure-shared.asciidoc[tag=param-grok-pattern]
+include::{es-repo-dir}/text-structure/apis/find-structure-shared.asciidoc[tag=param-ecs-compatibility]
+include::{es-repo-dir}/text-structure/apis/find-structure-shared.asciidoc[tag=param-quote]
+include::{es-repo-dir}/text-structure/apis/find-structure-shared.asciidoc[tag=param-should-trim-fields]
+include::{es-repo-dir}/text-structure/apis/find-structure-shared.asciidoc[tag=param-timeout]
+include::{es-repo-dir}/text-structure/apis/find-structure-shared.asciidoc[tag=param-timestamp-field]
+include::{es-repo-dir}/text-structure/apis/find-structure-shared.asciidoc[tag=param-timestamp-format]
+
+[discrete]
+[[find-field-structure-examples]]
+== {api-examples-title}
+
+[discrete]
+[[find-field-structure-example]]
+=== Analyzing Elasticsearch log files
+
+Suppose you have a list of Elasticsearch logs messages in an index.
+You can analyze them with the `find_field_structure` endpoint as follows:
+
+[source,console]
+----
+POST _bulk?refresh=true
+{"index":{"_index":"test-logs"}}
+{"message":"[2024-03-05T10:52:36,256][INFO ][o.a.l.u.VectorUtilPanamaProvider] [laptop] Java vector incubator API enabled; uses preferredBitSize=128"}
+{"index":{"_index":"test-logs"}}
+{"message":"[2024-03-05T10:52:41,038][INFO ][o.e.p.PluginsService     ] [laptop] loaded module [repository-url]"}
+{"index":{"_index":"test-logs"}}
+{"message":"[2024-03-05T10:52:41,042][INFO ][o.e.p.PluginsService     ] [laptop] loaded module [rest-root]"}
+{"index":{"_index":"test-logs"}}
+{"message":"[2024-03-05T10:52:41,043][INFO ][o.e.p.PluginsService     ] [laptop] loaded module [x-pack-core]tive methods for [MacOS]"}
+{"index":{"_index":"test-logs"}}
+{"message":"[2024-03-05T10:52:41,043][INFO ][o.e.p.PluginsService     ] [laptop] loaded module [x-pack-redact]API enabled; uses preferredBitSize=128"}
```

Comment:
It looks like this was created from one set of logs printed on top of another set.

```suggestion
{"message":"[2024-03-05T10:52:41,043][INFO ][o.e.p.PluginsService     ] [laptop] loaded module [x-pack-core]"}
{"index":{"_index":"test-logs"}}
{"message":"[2024-03-05T10:52:41,043][INFO ][o.e.p.PluginsService     ] [laptop] loaded module [x-pack-redact]"}
```

Reply:
very sharp... no clue how that happened 🤷